src/features/CourtCaseList/CourtCaseListEntry.tsx

```diff
@@ -0,0 +1,184 @@
+import ConditionalRender from "components/ConditionalRender"
+import DateTime from "components/DateTime"
+import { GridRow, Link, Table } from "govuk-react"
+import Image from "next/image"
+import { useRouter } from "next/router"
+import { encode } from "querystring"
+import { useState } from "react"
+import { createUseStyles } from "react-jss"
+import CourtCase from "services/entities/CourtCase"
+import User from "services/entities/User"
+import { deleteQueryParamsByName } from "utils/deleteQueryParam"
+import getTriggerWithDescription from "utils/formatReasons/getTriggerWithDescription"
+import groupErrorsFromReport from "utils/formatReasons/groupErrorsFromReport"
+import { displayedDateFormat } from "utils/formattedDate"
+import {
+  filterUserNotes,
+  first100CharsOfMostRecentNote,
+  getMostRecentNote,
+  validatedMostRecentNoteDate
+} from "./CourtCaseListEntryHelperFunction"
+import { NotePreview, NotePreviewButton } from "./NotePreviewButton"
+import LockedByTag from "./tags/LockedByTag"
+import ResolvedTag from "./tags/ResolvedTag"
+import UrgentTag from "./tags/UrgentTag"
+
+const useStyles = createUseStyles({
+  caseDetailsRow: {
+    verticalAlign: "top",
+    borderColor: "unset"
+  },
+  triggersRow: {
+    verticalAlign: "top",
+    backgroundColor: "#f3f2f1" // GDS light-grey color
+  },
+  flexBox: {
+    display: "flex"
+  },
+  notesRow: {
+    borderTop: "white solid"
+  }
+})
+
+interface Props {
+  courtCase: CourtCase
+  currentUser: User
+}
+
+const CourtCaseListEntry: React.FC<Props> = ({ courtCase, currentUser }: Props) => {
+  const {
+    errorId,
+    courtDate,
+    ptiurn,
+    defendantName,
+    courtName,
+    triggers,
+    errorReport,
+    isUrgent,
+    errorLockedByUsername,
+    triggerLockedByUsername,
+    resolutionTimestamp,
+    notes
+  } = courtCase
+  const classes = useStyles()
+  const { basePath, query } = useRouter()
+  let searchParams = new URLSearchParams(encode(query))
+  searchParams = deleteQueryParamsByName(["unlockException", "unlockTrigger"], searchParams)
+  const caseDetailsPath = (id: number) => `${basePath}/court-cases/${id}`
+  const unlockCaseWithReasonPath = (reason: "Trigger" | "Exception", caseId: string) => {
+    searchParams.append(`unlock${reason}`, caseId)
+    return `${basePath}/?${searchParams}`
+  }
+  const canUnlockCase = (lockedUsername: string): boolean => {
+    return currentUser.groups.includes("Supervisor") || currentUser.username === lockedUsername
+  }
+
+  const exceptions = groupErrorsFromReport(errorReport)
+  const [showPreview, setShowPreview] = useState(false)
+  const userNotes = filterUserNotes(notes)
+  const numberOfNotes = userNotes.length
+  const mostRecentUserNote = getMostRecentNote(userNotes)
+
+  const triggerRow =
```

Comment:
Why wouldn't this be a separate component? It's gut quite a few lines of html.

Reply:
This has not been changed, we just broke it out into a new component to simplify CourtCaseList and fix an error we were having of using useState() within a loop.

It could be abstracted into a separate component but I do not think it is in scope of this notePreview ticket

Reply:
Sounds good! I'm resolving this comment.